token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -450,139 +452,329 @@ fn process_transfer(
     let mint_info = next_account_info(account_info_iter)?;
     let instructions_sysvar_info = next_account_info(account_info_iter)?;
     let authority_info = next_account_info(account_info_iter)?;
-    let authority_info_data_len = authority_info.data_len();
 
     check_program_account(mint_info.owner)?;
     let mint_data = &mint_info.data.borrow_mut();
     let mint = StateWithExtensions::<Mint>::unpack(mint_data)?;
     let ct_mint = mint.get_extension::<ConfidentialTransferMint>()?;
 
+    let previous_instruction =
+        get_instruction_relative(proof_instruction_offset, instructions_sysvar_info)?;
+
+    if let Ok(transfer_fee_config) = mint.get_extension::<TransferFeeConfig>() {
+        // mint is extended for fees
+        let proof_data = decode_proof_instruction::<TransferWithFeeData>(
+            ProofInstruction::VerifyTransfer,
+            &previous_instruction,
+        )?;
+
+        if proof_data.transfer_with_fee_pubkeys.auditor != ct_mint.auditor_pk {
```

Comment:
It'd be nice if the lhs and rhs fields matched to enhance readability: `. auditor` vs `.auditor_pk`
This could be potentially done as a follow-up PR though

Reply:
Great! I changed `auditor_pk` in the mint extension to `auditor_pubkey` for now. I will update the `auditor` in the zk-token-sdk side in the next batch of updates for `0.7.0`.